File: Arrays/SmallestSubarrayWithSumGreater/SmallestSubarrayWithSumGreater.js
function smallestSubWithSum(arr,n,k){
    //code here
    let sum=arr[0], right=0, left=0, len=Number.MAX_VALUE, temp=0;
while(right<n){
     if(sum<=k){
        right++;
        sum += arr[right];
    }
    if(sum>k){
        temp = right-left;
        if(temp<len){
            len = temp;
        }
        sum-=arr[left];
        left++;
    }
}
return len+1;
}

let arr = [1, 4, 45, 6, 10, 19];
let x = 51;
console.log(smallestSubWithSum(arr, x));

File: Arrays/SmallestSubarrayWithSumGreater/SmallestSubarrayWithSumGreater.go
package main

import "fmt"

func method1(arr []int, k int) int {
	n := len(arr)
	count := n
	sum := 0
	j := 0
	for i := 0; i < n; {
		for sum <= k && i < n {
			sum += arr[i]
			i++
		}
		for sum > k && j < n {
			sum -= arr[j]
			if i-j < count {
				count = i - j
			}
			j++
		}
	}
	return count
}

func main() {
	arr := []int{1, 4, 45, 6, 0, 19}
	k := 51
	fmt.Println(method1(arr, k))
}
